util/resolver/retryhandler/retry.go

```diff
@@ -59,8 +59,8 @@ func retryError(err error) bool {
 	if errors.Is(err, io.EOF) || errors.Is(err, syscall.ECONNRESET) || errors.Is(err, syscall.EPIPE) || errors.Is(err, net.ErrClosed) {
 		return true
 	}
-	// catches TLS timeout or other network-related temporary errors
-	if ne, ok := errors.Cause(err).(net.Error); ok && ne.Temporary() {
+	// catches TLS timeout
+	if ne, ok := errors.Cause(err).(net.Error); ok && ne.Timeout() {
```

Comment:
While `Temporary()` is marked deprecated it doesn't mean that `Timeout()` is the direct replacement. https://github.com/golang/go/issues/45729 Looks like we leave many previous cases unhandled now. If we don't want to skip this warning then we might need to define our own definition for temporary error.

Reply:
Indeed, I was talking with @thaJeztah about this one and some errors may continue to propagate through the "temporary" status.